classes/Context.php

```diff
@@ -422,49 +422,15 @@ public function getTranslatorFromLocale($locale)
             (new Filesystem())->remove($cache_file);
         }
 
-        $adminContext = defined('_PS_ADMIN_DIR_');
-        $translator->addLoader('xlf', new XliffFileLoader());
-
-        $sqlTranslationLoader = new SqlTranslationLoader();
-        if (null !== $this->shop) {
-            $sqlTranslationLoader->setTheme($this->shop->theme);
-        }
-
-        $translator->addLoader('db', $sqlTranslationLoader);
-        $notName = $adminContext ? '^Shop*' : '^Admin*';
-
-        $finder = Finder::create()
-            ->files()
-            ->name('*.' . $locale . '.xlf')
-            ->notName($notName)
-            ->in($this->getTranslationResourcesDirectories());
+        // only way to reset the catalogue
+        $translator->addLoader('Array', new ArrayLoader());
```

Comment:
Maybe you should put this in the `TranslatorLanguageLoader::loadLanguage` method?